File: Python3/_1221.py
# Greedy algo: Track sum. Every time we arrive at
# equal # of 'L's and 'R's, ++numBalanced

class Solution:
    def balancedStringSplit(self, s: str) -> int:
        numBalanced = 0
        sum_ = 0

        for char in s:
        	sum_ += 1 if char == 'L' else -1;
        	if sum_ == 0:
        		numBalanced += 1

        return numBalanced

File: Cpp/_1221.cpp
// Greedy algo: Track sum. Every time we arrive at
// equal # of 'L's and 'R's, ++numBalanced

class Solution {
public:
int balancedStringSplit(string s) {
    int sum = 0, numBalanced = 0;
    for (const auto& c : s) {
        sum += (c == 'L' ? 1 : -1);
        if (sum == 0) ++numBalanced;
    }
    return numBalanced;        
}
};